crates/chia-protocol/src/block_record.rs

```diff
@@ -171,38 +164,38 @@ impl BlockRecord {
         py: Python<'a>,
         constants: &Bound<'_, PyAny>,
     ) -> PyResult<Bound<'a, PyAny>> {
-        ChiaToPython::to_python(&self.sp_sub_slot_total_iters_impl(py, constants)?, py)
+        ChiaToPython::to_python(&self.sp_sub_slot_total_iters_impl(constants)?, py)
     }
 
     fn ip_sub_slot_total_iters<'a>(
         &self,
         py: Python<'a>,
         constants: &Bound<'_, PyAny>,
     ) -> PyResult<Bound<'a, PyAny>> {
-        ChiaToPython::to_python(&self.ip_sub_slot_total_iters_impl(py, constants)?, py)
+        ChiaToPython::to_python(&self.ip_sub_slot_total_iters_impl(constants)?, py)
     }
 
     fn sp_iters<'a>(
         &self,
         py: Python<'a>,
         constants: &Bound<'_, PyAny>,
     ) -> PyResult<Bound<'a, PyAny>> {
-        ChiaToPython::to_python(&self.sp_iters_impl(py, constants)?, py)
+        ChiaToPython::to_python(&self.py_sp_iters_impl(constants)?, py)
     }
 
     fn ip_iters<'a>(
         &self,
         py: Python<'a>,
         constants: &Bound<'_, PyAny>,
     ) -> PyResult<Bound<'a, PyAny>> {
-        ChiaToPython::to_python(&self.ip_iters_impl(py, constants)?, py)
+        ChiaToPython::to_python(&self.py_ip_iters_impl(constants)?, py)
     }
 
     fn sp_total_iters<'a>(
         &self,
         py: Python<'a>,
         constants: &Bound<'_, PyAny>,
     ) -> PyResult<Bound<'a, PyAny>> {
-        ChiaToPython::to_python(&self.sp_total_iters_impl(py, constants)?, py)
+        ChiaToPython::to_python(&self.sp_total_iters_impl(constants)?, py)
```

Comment:
it's not clear to me why we need these extra wrapper functions. I would expect we could have pure rust member functions and (when enabled) python wrappers that call the rust functions. This appears to add 2 layers of python wrapper around the rust functions